### PR TITLE
Insertion and removal tests

### PR DESCRIPTION
**Pull request by sambley**
Noticing failures with the removal of items at random places. Added a test to reproduce the issue, since the issue is random it may require the "cargo +nightly bench" to be run a few times to be able to see the failure.

### CODE REVIEW

**Comment by Amanieu** [comment]
Can you change this into a `#[test]` and change the random-number generator to a deterministic one (such as `XorShiftRng`)? This should allow the test to reliably reproduce the issue.

**Comment by Amanieu** [comment]
This is fixed in #26

File: Cargo.toml [hunk 1]
```diff
@@ -13,6 +13,7 @@ categories = ["data-structures", "no-std"]
 scopeguard = { version = "0.3", default-features = false }
 byteorder = { version = "1.0", default-features = false }
 serde = { version = "1.0", default-features = false, optional = true }
+rand = "0.5.1"
```
Comment (Amanieu) [inline Cargo.toml]: This should be in dev-dependencies since it is only used for tests.

File: benches/bench.rs [hunk 2]
```diff
@@ -58,6 +61,27 @@ fn grow_by_insertion(b: &mut Bencher) {
...
+            assert_eq!(m.insert(tx[i].clone()), true);
+        }
+        for i in 0..4096 {
+            println!("removing {} {:?}", i, tx[i]);
+            assert_eq!(m.remove(&tx[i]), true);
+        }
+    });
+}
+
+
```
Comment (Amanieu) [inline benches/bench.rs]: Delete this since we have the version in tests/set.rs now.

**Comment by Amanieu** [comment]
Oops, didn't mean to close this.

**Comment by sambley** [comment]
Thanks, that worked.

**Comment by hcpl** [comment]
@sambley I think you didn't mean to close this? https://github.com/Amanieu/hashbrown/pull/26 only fixes the test failures you mentioned here, the tests themselves haven't been merged yet.

File: benches/bench.rs (hunk 2 above)
Comment (Amanieu) [reply benches/bench.rs]: This is not resolved. You should revert your changes to the benchmarks file and only keep the file in `tests`.

**Pull request closed without merging**